simulationworkflowschema/equation_of_state.py

```diff
@@ -158,8 +252,52 @@ def normalize(self, archive, logger):
             self.results = EquationOfStateResults()
             self.outputs.append(Link(name=WORKFLOW_RESULTS_NAME, section=self.results))
 
+        try:
+            task_archives = [task.task.m_root() for task in self.tasks]
+            assert all(
+                isinstance(task_archive.workflow2, SinglePoint)
+                for task_archive in task_archives
+            )
+        except Exception:
+            logger.warning(
+                'Not all tasks are SinglePoints or failed to retrieve task archives. EOS workflow may be incomplete or incorrect.'
+            )
+
+        for task in self.tasks:
+            flag_input_structure = False
+            for input in task.inputs:
+                # logger.warning(
+                #     f'input_structure.m_def: {input_structure["section"].m_def}'
+                # )
+                # logger.warning(f'input.section.m_def: {input.section.m_def}')
+                if input.section.m_proxy_value == input_proxy_value:
```

Comment:
i think you can directly compare sections, or does it fail when a section needs resolution?

Reply:
maybe do something like ```
sections = [input.section for input in task.inputs]
if input_section in sections:
    task.inputs[sections.index(input_section)].name = input_section.name
```

Reply:
I tested this, but `if input_section in sections:` doesn't work, any other ideas for comparing sections?

Reply:
can you probably inspect what you have in sections and what your input_section is? i just thought that you can directly compare sections but maybe they will not be the same if resolved. i do not know of any other `ids` that you can use to compare sections.